genienlp/models/transformer_lstm.py

```diff
@@ -56,17 +58,28 @@ def __init__(self, config=None, *inputs, args, vocab_sets, tasks, save_directory
         encoder_embeddings = args.pretrained_model
         config = AutoConfig.from_pretrained(encoder_embeddings, cache_dir=args.embeddings)
         args.dimension = config.hidden_size
-        self.numericalizer = TransformerNumericalizer(encoder_embeddings,
-                                                      max_generative_vocab=args.max_generative_vocab,
-                                                      cache=args.embeddings,
-                                                      preprocess_special_tokens=args.preprocess_special_tokens)
+        self.numericalizer = TransformerNumericalizer(encoder_embeddings, args, max_generative_vocab=args.max_generative_vocab)
+        
+        self.numericalizer.get_tokenizer(save_directory)
         self.init_vocab_from_data(vocab_sets, tasks, save_directory)
 
         logger.info(f'Initializing encoder and decoder embeddings')
-        if save_directory is not None:
-            self.encoder_embeddings = AutoModel.from_config(config)
+        
+        if args.do_ner:
+            if type(config) == BertConfig:
+                self.encoder_embeddings = BertModelForNER(config, args.num_db_types, args.db_unk_id).from_pretrained(
+                    encoder_embeddings, num_db_types=args.num_db_types, db_unk_id=args.db_unk_id, cache_dir=args.embeddings)
+            elif type(config) == XLMRobertaConfig:
+                self.encoder_embeddings = XLMRobertaModelForNER(config, args.num_db_types, args.db_unk_id).from_pretrained(
+                    encoder_embeddings, num_db_types=args.num_db_types, db_unk_id=args.db_unk_id, cache_dir=args.embeddings)
```

Comment:
This code needs to handle save_directory the same way as the other branch (ie, not call from_pretrained if save_directory is set)